iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/AddRegionPeerProcedure.java

```diff
@@ -92,6 +94,7 @@ protected Flow executeFromState(ConfigNodeProcedureEnv env, AddRegionPeerState s
           setNextState(AddRegionPeerState.DO_ADD_REGION_PEER);
           break;
         case DO_ADD_REGION_PEER:
+          handler.updateRegionCache(consensusGroupId, destDataNode, RegionStatus.Adding);
```

Comment:
Why should update RegionStatus to Adding twice?

Reply:
This is for robustness. The procedure may recover to different state.